ts/screens/wallet/payment/ScanQRCodeScreen.tsx

```diff
@@ -106,15 +134,41 @@ const styles = StyleSheet.create({
   }
 });
 
-export class QRcodeAcquisitionByScannerScreen extends React.Component<
-  Props,
-  never
-> {
+const QRCODE_SCANNER_REACTIVATION_TIME_MS = 2000;
+
+const rptIdFromQrCodeString = (qrCodeString: string): Either<t.Errors, RptId> =>
+  PaymentNoticeQrCodeFromString.decode(qrCodeString).chain(
+    rptIdFromPaymentNoticeQrCode
+  );
+
+class ScanQRCodeScreen extends React.Component<Props, never> {
   private goBack() {
     this.props.navigation.goBack();
   }
 
+  private qrCodeRead = (data: string) => {
+    const rptId = rptIdFromQrCodeString(data);
+    const paymentNotice = PaymentNoticeQrCodeFromString.decode(data);
+    if (rptId.isRight() && paymentNotice.isRight()) {
+      // successful conversion to RptId
+      this.props.showTransactionSummary(
+        rptId.value,
+        paymentNotice.value.amount
+      );
+    } // else toast stating that QR code is invalid
```

Comment:
`TODO` or we will forget about this - are you going to implement the error cases here or in a different PR? in that case we need stories...